### DISCUSSION TITLE
Add support for exp and iat on Request Object

### DISCUSSION THREAD

**Discussion by pasqualebarbaromind** [OpenIDC/mod_auth_openidc#1136] [Ideas]
The OIDCRequestObject config doesn't provide any option to add the exp (expiration) and iat (issued at) claims.
Since these claims in request object are adopted by some specifications such as https://openid.net/specs/openid-federation-1_0.html#name-using-a-request-object it would be great to have them added in the OIDCRequestObject configuration.

**Comment by zandbelt** [comment]
good point indeed, we'll add `iat` and a "now+30s" `exp` value by default and have a configurable "ttl" setting that would be the input for a customized `exp`

**Comment by zandbelt** [comment]
added in 474a78c4098e8af171b234af33c3822d8ba788c6

**Comment by pasqualebarbaromind** [comment]
Wow, I wasn't expecting such responsiveness!
Thanks for your fast reply and implementation :)
I have a little suggestion to improve what is written in [auth_openidc.conf](https://github.com/OpenIDC/mod_auth_openidc/blob/master/auth_openidc.conf) regarding this new option: maybe it would be useful to add a mention to the iat claim, since only the exp is mentioned, and users would find the iat claim without knowing where it comes from

**Reply by zandbelt** [reply]
there's other required stuff that is not explicitly mentioned in the config file (because it is simply not configurable) and the ChangeLog and Release notes reflect changes between releases

**Reply by pasqualebarbaromind** [reply]
Right, better to stay consistent with other configs. Thanks again

**Comment by zandbelt** [comment]
this is now in 2.4.15 https://github.com/OpenIDC/mod_auth_openidc/releases/tag/v2.4.15
